### ISSUE TITLE
Please Add Typescript support

### ISSUE THREAD

**Issue by DevYeThiha** [stephenscaff/react-animated-cursor#28]

**Comment by Thremulant** [comment]
hey @MightyCryptowall  I converted this package to TS and published **FOR TESTING** purposes, can you please give it a try? 
It has been working on a couple of CRA projects with and without TS, but I'd like other users to try it. 

Here's the link: [react-animated-cursor-ts](https://www.npmjs.com/package/react-animated-cursor-ts)

**Comment by MightyCryptowall** [comment]
there is some error in SSR
<img width="982" alt="Screen Shot 2022-09-09 at 13 31 26" src="https://user-images.githubusercontent.com/52656718/189291002-7b435f2d-7656-4bdf-b4a5-0847b4512afd.png">

**Comment by Thremulant** [comment]
@MightyCryptowall expected, as the files no longer have a `default` module. Try it using the `named export`:

````javascript
const AnimatedCursor = dynamic(() => import('react-animated-cursor-ts')
  .then((comp) => comp.AnimatedCursor), {
  ssr: false
});
````
[Dynamic Import with named export - NextJS Documentation](https://nextjs.org/docs/advanced-features/dynamic-import#with-named-exports)

**Comment by MightyCryptowall** [comment]
@Thremulant, It works. thx

**Comment by stephenscaff** [comment]
Updated to TS Ya'll. Just published v 2.7.0 with has TS support.
 
I made various updates before migrating to TS (most importantly changing how the scale animation worked to address a known safari bug causing blurring), in addition to upgrading all the deps to latest (including moving to rollup v3), so couldn't just accept @Thremulant's PR as is. But, used that as my starting point, so super grateful for the help. 

Give it a shot. Let me know if you have any issues / suggestions.

Thanks.
